src/WorldHandler.cpp

```diff
@@ -57,10 +57,14 @@ void WorldHandler::handleRefboxPackets(proto::SSL_Referee &ref_packet) const {
     }
 }
 
-void WorldHandler::handleVisionPackets(proto::SSL_WrapperPacket &vision_packet) const {
+void WorldHandler::handleVisionPackets(proto::SSL_WrapperPacket &vision_packet) {
     while (vision_client && vision_client->receive(vision_packet)) {
         if (vision_packet.has_detection()){
-            KF->newFrame(vision_packet.detection());
+            double time=vision_packet.detection().t_capture();
+            if (time>lastPacketTime){
+                lastPacketTime=time;
+            }
+            KF->addFrame(vision_packet.detection());
```

Comment:
This just keeps track of the highest  packet capture time. Packets sometimes arrive asynchronously by detection since a packet sometimes takes longer to transmit or to analyze through SSL-vision (e.g. a packet with timestamp 1.0 might arrive before a packet with timestamp 0.99).

This whole construction is a sign of other problems with timing in our system though, which mainly exists because we do not know delays etc. and haven't build our system around it yet.